x/leverage/keeper/msg_server.go

```diff
@@ -367,14 +367,24 @@ func (s msgServer) MaxBorrow(
 	// but not this token or any of their borrows, error
 	// will be nil and the resulting value will be what
 	// can safely be borrowed even with missing prices.
-	maxBorrow, err := s.keeper.maxBorrow(ctx, borrowerAddr, msg.Denom)
+	userMaxBorrow, err := s.keeper.userMaxBorrow(ctx, borrowerAddr, msg.Denom)
+	if err != nil {
+		return nil, err
+	}
+	if userMaxBorrow.IsZero() {
+		return &types.MsgMaxBorrowResponse{Borrowed: coin.Zero(msg.Denom)}, nil
+	}
+
+	// Get the max available to borrow from the module
+	maxBorrow, err := s.keeper.moduleMaxBorrow(ctx, userMaxBorrow)
```

Comment:
Maybe keep the comparison to `userMaxBorrow` outside of the `moduleMaxBorrow` function, to simplify its signature and potentially allow it to be used for queries about the module (independent of user) in the future.